### PR TITLE
Favor docker compose v2 in ansible tasks

### LINKED ISSUE #1834: [BUG]: Ansible playbook restore_locally.yml errors out
### Steps to reproduce.

To reproduce this, you must have a backup of an instance locally in the ansible/backups directory. This is stored when running the playbook ansible/backup.yml.

* With a local backup, run the playbook:
```
ansible-playbook ansible/restore_locally.yml
```
* After choosing a playbook, you will encounter an error trying to use pip install docker-compose:
```
atal: [127.0.0.1]: FAILED! => {"changed": false, "cmd": ["/usr/bin/python", "-m", "pip.__main__", "install", "docker-compose"], "msg": "\n:stderr: error: externally-managed-environment\n\n× This environment is externally managed\n╰─> To install Python packages system-wide, try 'pacman -S\n    python-xyz', where xyz is the package you are trying to\n    install.\n    \n    If you wish to install a non-Arch-packaged Python package,\n    create a virtual environment using 'python -m venv path/to/venv'.\n    Then use path/to/venv/bin/python and path/to/venv/bin/pip.\n    \n    If you wish to install a non-Arch packaged Python application,\n    it may be easiest to use 'pipx install xyz', which will manage a\n    virtual environment for you. Make sure you have python-pipx\n    installed via pacman.\n\nnote: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.\nhint: See PEP 668 for the detailed specification.\n"}
```

Removing this step, throws the error:

```
fatal: [127.0.0.1]: FAILED! => {"changed": false, "msg": "Configuration error - kwargs_from_env() got an unexpected keyword argument 'ssl_version'"}
```

this is related to https://github.com/tactilenews/100eyes/issues/1768

There is now a `community.docker.docker_compose_v2` module that we can use.

### PR DESCRIPTION
**Pull request by mattwr18**
Fixes #1834 

I've tested running the `ansible/restore_locally` and `ansible/site` playbooks and nothing appears to be broken.

### CODE REVIEW

File: ansible/roles/installation/tasks/main.yml [hunk 1]
```diff
@@ -56,23 +56,23 @@
...
     project_src: /home/ansible
-    build: no
-    debug: yes
-    pull: yes
-    restarted: yes
+    recreate: always
+    build: never
+    pull: always
+    state: present
+    wait: true
```
Comment (roschaefer) [inline ansible/roles/installation/tasks/main.yml:65]: @mattwr18 this is a nice new feature that will ensure that all containers are healthy before proceeding. In our pair-programming today we were facing the issue of silent failures.

File: ansible/site.yml [hunk 2]
```diff
@@ -21,14 +21,11 @@
     - role: gantsign.inotify
       inotify_max_user_watches: 524288
     - role: geerlingguy.pip
-      vars:
-        pip_install_packages:
-          - name: "docker < 7.0.0"
-          - name: docker-compose
-          - name: passlib
+      pip_install_packages:
+        - name: passlib
```
Comment (roschaefer) [inline ansible/site.yml:25]: `passlib` is required for `htpasswd` role. The other two seem not required.

File: ansible/site.yml [hunk 3]
```diff
@@ -21,14 +21,11 @@
...
     - role: geerlingguy.docker
+      docker_packages_state: latest
```
Comment (roschaefer) [inline ansible/site.yml:27]: I'm willing to keep this variable. Opinions/thoughts?

**Review by roschaefer — APPROVED** [review]
docker

**Comment by mattwr18** [comment]
Also checked your changes @roschaefer and they look good...

**Pull request merged**